### PR TITLE
Error in odh-model-controller logs on deletion of ModelMesh isvc

### PR DESCRIPTION
**Pull request by vaibhavjainwiz**
cherry pick https://github.com/opendatahub-io/odh-model-controller/pull/153

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: config/rbac/role.yaml [hunk 1]
```diff
@@ -4,13 +4,24 @@ kind: ClusterRole
 metadata:
   name: odh-model-controller-role
 rules:
+- apiGroups:
```
Comment (aslakknutsen) [inline config/rbac/role.yaml:7]: This file is generated from the  kubebuilder rules in main.go. Any change to the RBAC rules should be applied to main.go. 

https://github.com/opendatahub-io/odh-model-controller/blob/f6fb33a819b8c91c8c08b9612ca8e9ad9282ed84/main.go#L56-L73

**Review by aslakknutsen — CHANGES_REQUESTED** [review]

File: config/rbac/role.yaml (hunk 1 above)
Comment (Jooho) [reply config/rbac/role.yaml:7]: @vaibhavjainwiz I agreed with @aslakknutsen . Line #73 should be updated.

**Review by Jooho — COMMENTED** [review]
updating manifests directly is not the right way because it will be regenerated by kubebuilder.

**Comment by israel-hdez** [comment]
@vaibhavjainwiz Is this still needed?

File: config/rbac/role.yaml (hunk 1 above)
Comment (spolti) [reply config/rbac/role.yaml:7]: @vaibhavjainwiz do you have bandwidth to do this?

**Comment by spolti** [comment]
closed as stale.

**Pull request closed without merging**